### PR TITLE
Add Github Pages deploy workflow

### LINKED ISSUE #27: Build release automation
Use our best GitOps principles to publish updates to the website after pull requests are merged. Netlify also supports preview URLs for open pull requests, if that's possible as well.

### PR DESCRIPTION
**Pull request by RedbackThomson**
Closes #27 

Create a Github action using the "peaceiris/actions-gh-pages" action step to deploy to Github pages. Uses the `contents: write` permission to gain write access to branches (through the `GITHUB_TOKEN` secret). 

Added a CNAME file for consistency with Github pages documentation.

### CODE REVIEW

File: CNAME [hunk 1]
```diff
@@ -0,0 +1 @@
+opengitops.dev
```
Comment (scottrigby) [inline CNAME:1]: Per https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-add-cname-file-cname I believe the existing file we have here is correct: https://github.com/open-gitops/website/blob/main/public/CNAME. Is this necessary So let's remove this duplicate file from the top level of your PR branch.

Looks like we could also add 
```yaml
with:
    cname: opengitops.dev
```
to the GitHub Pages action workflow instead. Either way should be fine

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - run: npm ci
+      - run: npm run build
+      - name: GitHub Pages action
+        uses: peaceiris/actions-gh-pages@v3
+        if: ${{ github.ref == 'refs/heads/main' }}
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./public
```
Comment (scottrigby) [inline .github/workflows/deploy.yml:38]: minor: needs newline at EOF
```suggestion
          publish_dir: ./public

```

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: Deploy Gatsby
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
+
```
Comment (scottrigby) [inline .github/workflows/deploy.yml:10]: minor: YAML max 1 consecutive blank line
```suggestion
```

**Review by scottrigby — CHANGES_REQUESTED** [review]
@RedbackThomson thanks for this PR!

Quick untested review lgtm 👀 👌 per https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-gatsby

Check out my CNAME comment. The line breaks are minor but may as well clean those up.

I'd say let's get this fixed and merge/test it! If there's any issue we can simply revert the PR and try again. But it would be really good to automate the deploy, as we're currently busy with other things and this will help the site run smoothly.

File: CNAME (hunk 1 above)
Comment (RedbackThomson) [reply CNAME:1]: Ah didn't catch that CNAME in public. Will remove my change - prefer to use that than bake it into the workflow, for flexibility.

**Comment by RedbackThomson** [comment]
Thanks @scottrigby . Updated!

**Comment by scottrigby** [comment]
@RedbackThomson nice! Just need to fix DCO (see https://github.com/open-gitops/website/pull/30/checks)

**Review by scottrigby — APPROVED** [review]

**Pull request merged**